serializer/seata-serializer-seata/src/main/java/io/seata/serializer/seata/protocol/transaction/AbstractBranchEndResponseCodec.java

```diff
@@ -52,7 +52,7 @@
             out.writeShort((short)0);
         }
         out.writeLong(branchId);
-        out.writeByte(branchStatus.getCode());
+        out.writeByte((branchStatus == null ? BranchStatus.Unknown : branchStatus).getCode());
```

Comment:
I think the main problem is that what kind of situation would let the branch status null in branch commit or branch rollback.

Reply:
because when branch commit or rollback, the branch status should not be null.

Reply:
OK, I'll find out why.

Reply:
> 
> 
> I think the main problem is that what kind of situation would let the branch status null in branch commit or branch rollback.

I found the cause of the problem and fixed it.